src/features/contracts/index.ts

```diff
@@ -164,8 +192,10 @@ export const synchronizeContracts = async (contractRepos: RepositoryInfo[]) => {
 		console.error(`Failed to fetch contracts, skipping...`, err.message);
 	}
 
-	await Promise.all(
-		['hw.device-type', 'arch.sw'].map(async (contractType) => {
+	// We don't have automatic dependency resolution, so the order matters here.
+	await Bluebird.map(
+		['arch.sw', 'hw.device-type'],
+		async (contractType) => {
```

Comment:
If you want concurrency of 1 you can just use a for loop:
```suggestion
	for (const contractType of ['arch.sw', 'hw.device-type']) {
```